src/app/components/ElectronicDelivery/ElectronicDelivery.jsx

```diff
@@ -42,26 +43,41 @@ class ElectronicDelivery extends React.Component {
     this.requireUser();
   }
 
-  /**
-   * requireUser()
-   * Redirects the patron to OAuth log in page if he/she hasn't been logged in yet.
-   *
-   * @return {Boolean}
+  /*
+   * componentDidUpdate()
+   * After the component updates, if there are errors then the DOM for the error box message
+   * is rendered. Since it exists, it should be focused so that the patron can get a better
+   * idea of what is wrong.
    */
-  requireUser() {
-    if (this.state.patron && this.state.patron.id) {
-      return true;
+  componentDidUpdate() {
+    if (this.refs['nypl-form-error']) {
+      ReactDOM.findDOMNode(this.refs['nypl-form-error']).focus();
     }
+  }
 
-    const fullUrl = encodeURIComponent(window.location.href);
-
-    window.location.replace(`${appConfig.loginUrl}?redirect_uri=${fullUrl}`);
+  /*
+   * getRaisedErrors(raiseError)
+   * There's a set list of required inputs in the EDD form. If the key errors from the form
+   * are found in the set list, it will render those errors. This is meant to be an
+   * aggregate list that is displayed at the top of the form.
+   * @param {object} raiseError An object with the key/value pair of input elements in the
+   *   EDD form that have incorrect input.
+   * @return {object}
+   */
+  getRaisedErrors(raiseError) {
+    const headlineError = {
+      emailAddress: 'Email Address',
+      chapterTitle: 'Chapter / Article Title',
+      startPage: 'Starting Page Number',
+      endPage: 'Ending Page Number',
+    };
 
-    return false;
-  }
+    const raisedErrors = [];
+    _mapObject(raiseError, (val, key) => {
```

Comment:
True. I didn't write this so it's actually not part of my PR. If you check the `render` function, line 201, this function will only get called if `raiseError` is defined. I can add another one just to be safe.